packages/core/docs/performance/optimizing-javascript.md

```diff
@@ -0,0 +1,68 @@
+
+There is also a vendor bundle with common parts shared between multiple bundles. For example, if Home and Product pages use the same Navigation component, it will go to the vendor bundle and be downloaded only once.
+
+Nuxt.js does code-splitting out of the box by creating separate bundles for every page/route. You can control this behavior using the [build.splitChunks](https://nuxtjs.org/docs/configuration-glossary/configuration-build/#splitchunks) property in `nuxt.config.js` file.
+
+## Avoid serving polyfills to modern browsers :ledger:
+
+Polyfills and transforms enable you to use new JavaScript features in a legacy browser. However, they are unnecessary in modern browsers, making the bundle bigger and impacting the performance.
+
+### Modern mode
+
+The Nuxt.js has a [--modern](https://nuxtjs.org/docs/configuration-glossary/configuration-modern/) parameter that you can use with the `nuxt build` command to create two bundles:
+
+* "legacy" bundle for older browsers,
+* "modern" bundle for evergreen browsers.
+
+Browsers will load only one of them, depending on whether it supports ES modules or not.
+
+### Configure Babel
+
+Babel is a toolchain used to converting modern JavaScript code into a backwards compatible version for current and older browsers or environments.
+
+Nuxt.js includes it out of the box. You can control its behavior using the [build.babel](https://nuxtjs.org/docs/configuration-glossary/configuration-build/#babel) property in `nuxt.config.js` file.
+
+The default targets are:
+
+* `ie: '9'` for the legacy bundle.
+* `esmodules:true` for the modern bundle.
+* `node: 'current'` for the server bundle.
+
+## Avoid adding third-party scripts :ledger:
+
+Third-party code can significantly impact the performance, and the best thing you can do is not to add them to your page at all. However, if you have to, there are some tricks to reduce the performance impact on your application.
+
+* Load scripts with the `async` or `defer` attribute to avoid blocking document parsing.
+* Self-host the script if the third-party server is slow.
+* Remove the script if it doesn't add clear value to your site.
+* Use the `rel=preconnect` or `rel=dns-prefetch` attributes in `<link>` to do a DNS lookup for domains hosting third-party scripts.
+* Lazy load third-party resources [with facades](https://web.dev/third-party-facades/?utm_source=lighthouse&utm_medium=devtools).
+* Move third-party scripts to Web worker using, for example, the [@nuxtjs/partytown](https://github.com/nuxt-community/partytown-module) module.
+
+### Avoid Google Tag Manager
+
+Google Tag Manager is often used to add additional scripts, styling, and elements or hide/show content on the page. This can lead to Cumulate Layout Shifts, extra Total Blocking Time, a bigger number of requests and their weight, or even rerendering the whole page.
```

Comment:
This is usually not decision made by dev team, perhaps suggest how to use it with less impact on performance and suggest smaller alternatives